d4s/src/main/scala/d4s/codecs/D4SEncoder.scala

```diff
@@ -49,7 +52,7 @@ object D4SEncoder {
       ctx.parameters.map {
         p =>
           p.label -> p.typeclass.encode(p.dereference(item))
-      }.toMap
+      }.toMap.filter(_._2.nonEmpty).view.mapValues(_.get).toMap
   }
```

Comment:
@YaroslavSahach do you wrap everything in Option to filter `None` values here?
If so, can we avoid that? I think you can check if attribute is null or not like this:
```scala
 ctx.parameters.map {
        p =>
          p.label -> p.typeclass.encode(p.dereference(item))
      }.toMap.view.filter {case (_, v) => !v.nul() }.toMap
```

Reply:
@vpodlnk good catch, thanks
I will correct that

Reply:
Yup, it's me who proposed to add Options because it's more expressive.

Reply:
Alright. I don't think it is bad, but IMO, it slightly tedious, we will wrap our value in Some(_) every time we will add a new encoder (which happens rarely, so probably not the case), plus we can do easily using AttriubteValue methods.

Anyway, if you guys think this will make code better/more concise, I am up for this.